src/test/java/org/opensearch/geospatial/ODFERestTestCase.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.apache.http.Header;
+import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.junit.After;
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.client.RestClient;
+import org.opensearch.client.RestClientBuilder;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.DeprecationHandler;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.test.rest.OpenSearchRestTestCase;
+
+/**
+ * ODFE integration test base class to support both security disabled and enabled ODFE cluster.
+ */
+public abstract class ODFERestTestCase extends OpenSearchRestTestCase {
```

Comment:
why we are naming it as ODFE? As per the context ODFE stands for OpenDistroForElasticSearch this was when we were using ElasticSearch. Let's name it something else like OpenSearchSecureRestTestCase

Reply:
Choose the name ODFERestTestCase as it is used in kNN plugin. 
Checked that other plugin named it as PluginRestTestCase. Will change the name as PluginRestTestCase to be consistent with other plugins if there is no other opinion.

Reply:
As K-NN was developed when we were using Elastic Search as the base for all our plugins. It's not migrated till now. But that is not the case with GeoSpatial plugin. 

We don't need to be consistent with other plugins, I feel as this class is encapsulating the security aspects so let's name it like that only. As we have GeospatialRestTestCase which is similar to PluginRestTestCase. It can create more confusion.

Reply:
Changed the name to OpenSearchSecureRestTestCase